restapi/src/main/java/io/stargate/web/docsapi/resources/ReactiveDocumentResourceV2.java

```diff
@@ -128,16 +336,20 @@ public void searchDoc(
               return reactiveDocumentService.findDocuments(
                   db, namespace, collection, where, fields, paginator, context);
             })
-        .map(
-            results -> {
-              if (raw == null || !raw) {
-                return Response.ok(results).build();
-              } else {
-                return Response.ok(results.getData()).build();
-              }
-            })
+        .map(rawDocumentHandler(raw))
         .safeSubscribe(
             AsyncObserver.forResponseWithHandler(
                 asyncResponse, ErrorHandler.EXCEPTION_TO_RESPONSE));
   }
+
+  private Function<DocumentResponseWrapper<? extends JsonNode>, Response> rawDocumentHandler(
+      @QueryParam("raw") @ApiParam(value = "Unwrap results", defaultValue = "false") Boolean raw) {
+    return results -> {
+      if (raw == null || !raw) {
```

Comment:
nit: invert the `if` condition for more readability, i.e. `raw != null && raw`?

Reply:
Done.